### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [PaddlePaddle/CINN#583]
[An extra null pointer check is not needed in functions](https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null "Do I need to check for null before delete p?") like the following.
- [Buffer::DeallocHost](https://github.com/PaddlePaddle/CINN/blob/85ab4981a38926dc5c1dbf672762cec335d2b857/cinn/runtime/buffer.h#L77 "Buffer::DeallocHost function")
- [Shape::Resize](https://github.com/PaddlePaddle/CINN/blob/85ab4981a38926dc5c1dbf672762cec335d2b857/cinn/runtime/buffer.cc#L29 "Shape::Resize function")

**Comment by luotao1** [comment]
@elfring Thanks very much for this suggestion, you can create a PR to fix it.

**Comment by enkilee** [comment]
https://github.com/PaddlePaddle/CINN/pull/1140

**Comment by elfring** [comment]
:thought_balloon: Would you become interested to [use a development tool like “clang-tidy” for corresponding source code adjustments](https://clang.llvm.org/extra/clang-tidy/checks/readability/delete-null-pointer.html "Deletion of unnecessary null pointer checks")?

**Comment by luotao1** [comment]
@elfring Yes, we have the plan to use `clang-tidy`, here is the [plan](https://github.com/PaddlePaddle/community/blob/master/pfcc/call-for-contributions/code_style/code_style_clang_tidy.md). However we don't have enough people and time to start this plan.
